sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/PartitionReceiver.cs

```diff
@@ -330,5 +392,28 @@ public virtual async Task CloseAsync(CancellationToken cancellationToken = defau
         ///
         [EditorBrowsable(EditorBrowsableState.Never)]
         public override string ToString() => base.ToString();
+
+        /// <summary>
+        ///   Creates an <see cref="TransportConsumer" /> to use for reading events.
+        /// </summary>
+        ///
+        /// <param name="consumerGroup">The name of the consumer group the consumer will be associated with.  Events will be read in the context of this group.</param>
+        /// <param name="partitionId">The identifier of the Event Hub partition from which events will be received.</param>
+        /// <param name="eventPosition">The position within the partition where the consumer should begin reading events.</param>
+        /// <param name="retryPolicy">The policy which governs retry behavior and try timeouts.</param>
+        /// <param name="trackLastEnqueuedEventProperties">Indicates whether information on the last enqueued event on the partition is sent as events are received.</param>
+        /// <param name="ownerLevel">The relative priority to associate with the link; for a non-exclusive link, this value should be <c>null</c>.</param>
+        /// <param name="prefetchCount">Controls the number of events received and queued locally without regard to whether an operation was requested.  If <c>null</c> a default will be used.</param>
+        ///
+        /// <returns>A <see cref="TransportConsumer" /> configured in the requested manner.</returns>
+        ///
+        internal virtual TransportConsumer CreateTransportConsumer(string consumerGroup,
+                                                                   string partitionId,
+                                                                   EventPosition eventPosition,
+                                                                   EventHubsRetryPolicy retryPolicy,
+                                                                   bool trackLastEnqueuedEventProperties,
+                                                                   long? ownerLevel,
+                                                                   uint? prefetchCount) =>
```

Comment:
Should this just take the options and then hold responsibility for applying them?  _(including the casting from int)_

Reply:
This method is only for testing purposes, so my idea here was making sure that we can capture the arguments the same way they are being passed to the `Connection` to avoid mistakes. I didn't want to give this responsibility to the testing method.

Reply:
Alternatively, we could do as you're suggesting, but we would need to add a test to make sure the method is passing the arguments as expected. I'll leave it as is unless you feel strongly about it.

Reply:
I leave it to your discretion, but since you're calling this from multiple constructors, it would allow you to centralize responsibility for mapping the options/input into the proper form for creating and give it more meaning.

Reply:
I'll do as you suggested then. I'll add a new test just to be safe.